terraform/modules/provisioners/dev_mode_provisioning.tf

```diff
@@ -2,7 +2,7 @@
 # Licensed under the Universal Permissive License v1.0 as shown at https://oss.oracle.com/licenses/upl.
 
 resource "null_resource" "dev_mode_provisioning" {
-  count = var.mode == "DEV" ? var.num_vm_instances : 0
+  count = var.mode == "DEV" && (var.is_bastion_instance_required || var.is_rms_private_endpoint_required) ? var.num_vm_instances : 0
```

Comment:
I thought dev mode is always going to work with bastion??

Reply:
Fixed it